Pipelines/templates/dotnet-test-job.yml

```diff
@@ -0,0 +1,44 @@
+parameters:
+# Job Name
+- name: jobName
+  type: string
+  default: 'dotnet_test'
+# Version of Dotnet SDK to use
+- name: dotnetVersion
+  type: string
+  default: '3.1.x'
+# Version of NuGet Tool to use
+- name: nugetVersion
+  type: string
+  default: '5.x'
+# List of paths to .csproj
+- name: projectPath
+  type: string
+  default: ''
+
+jobs:
+- job: ${{ parameters.jobName }}
+  displayName: Dotnet Test
+  pool:
+    vmImage: 'windows-latest'
+  steps:
+  - task: UseDotNet@2
+    displayName: Install Dotnet SDK
+    inputs:
+      packageType: 'sdk'
+      version: ${{ parameters.dotnetVersion }}
+  - task: NuGetToolInstaller@1
```

Comment:
why install nuget if we're using dotnet restore?

Reply:
Pretty sure 'dotnet restore' still leans on NuGet. This is for some past bugs we encountered where random packages weren't restoring correctly, and updating the NuGet tool before running 'dotnet restore' resolved the problem.